pkg/apis/autoscaling/annotation_validation.go

```diff
@@ -275,3 +275,37 @@ func validateInitialScale(config *autoscalerconfig.Config, m map[string]string)
 	}
 	return nil
 }
+
+func validateMaxInitialScale(config *autoscalerconfig.Config, m map[string]string) *apis.FieldError{
+
+	max,errs := getIntGE0(m,MaxScaleAnnotation)
+	initial,err := getIntGE0(m, InitialScaleAnnotation)
+
+	errs= errs.Also(err)
+    
+     if (initial<0){
+		errs=errs.Also(&apis.FieldError{
+			Message: fmt.Sprintf("initial-scale=%d should be greater than 0", initial),
+			Paths: []string{MaxScaleAnnotationKey,InitialScaleAnnotationKey},
+		})
+	 }else if !config.AllowZeroInitialScale && initial==0 {
+		errs=errs.Also(&apis.FieldError{
+			Message: fmt.Sprintf("initial-scale=%d not allowed by cluster", initial),
+			Paths: []string{MaxScaleAnnotationKey,InitialScaleAnnotationKey},
+		})
+	 }else if max<=0{
+		errs=errs.Also(&apis.FieldError{
+			Message: fmt.Sprintf("max-scale=%d should be greate than 0",max),
+			Paths: []string{MaxScaleAnnotationKey,InitialScaleAnnotationKey},
+		})
```

Comment:
I don't think these assertions are needed - we validate InitialScale here

https://github.com/knative/serving/blob/f1bd9296611ad2157be84f273dc627058be61a11/pkg/apis/autoscaling/annotation_validation.go#L265

We validate max scale here
https://github.com/knative/serving/blob/f1bd9296611ad2157be84f273dc627058be61a11/pkg/apis/autoscaling/annotation_validation.go#L178

We just need to perform validation if initial scale and max scale are present then do the comparison.

See here for examples:

https://github.com/knative/serving/blob/2579cc18142e7088d51be2bce62c22252f96150a/pkg/apis/autoscaling/annotation_validation.go#L178